edge-runtime/lib/response.ts

```diff
@@ -180,14 +182,16 @@ export const buildResponse = async ({
     }
 
     if (isDataReq) {
```

Comment:
As far as I can tell - this will only be truthy for pages router json requests
https://github.com/opennextjs/opennextjs-netlify/blob/a1d859dfa6868a668a6b84e117467a18c748439b/edge-runtime/lib/response.ts#L126-L127

App router RSC requests don't seem to have that header set so this _should_ be safe to do

Reply:
It looks like next uses `getRequestMeta(params.req, 'isNextDataReq')` for this? There's a chance we could do the same.
https://github.com/vercel/next.js/blob/canary/packages/next/src/server/request-meta.ts#L183-L205

Reply:
This was my initial attempt - https://github.com/opennextjs/opennextjs-netlify/pull/2734/commits/f5bb340270f656069efb84fe2abc3b7a91f37844 but to make it work, something in middleware handling (edge function) would need to signal to origin (lambda) that this meta data should be set - in that commit I just made the previous query param work again (if it was present I was setting that meta data), but this felt wrong given that Next.js explicitly removed support for that. I'm not exactly sure how it works on Vercel - I'm clearly missing something here, but I don't think attempting to set those meta tags directly is the right way to go - maybe figuring out how that meta data is meant to be set would be the best solution here - I'll try to look if I can figure that out

Reply:
Ok, so the part of code that is setting the meta data on requests (~https://github.com/vercel/next.js/blob/canary/packages/next/src/server/lib/router-utils/resolve-routes.ts) ... but it only happens when it is `_next/data` request to begin with, so because we were using canonical path before we never hit this code path and we just relied on fact that we could "externally" force it with query param.

So current fix (at least in terms of the approach of using data url instead of trying to add the internal meta data on request ourselves) seems like generally correct approach to me

Reply:
Thanks for looking into it even more deeply!